### DISCUSSION TITLE
How to paste programmatically? (Implementing a custom toolbar)

### DISCUSSION THREAD

**Discussion by gustavotoyota** [ueberdosis/tiptap#3129] [General Questions]
I would like to design a toolbar with functions like Copy, Cut and Paste.
Is there a way to execute these functions programmatically in Tiptap?
Or is there another way to implement a custom toolbar with these functions?

I can't use `document.execCommand('paste')` because it's deprecated.
I don't know if it's possible to do it with the Clipboard API.

Thanks.

**Comment by jlarmstrongiv** [comment]
While `document.execCommand` is deprecated, there are [no viable alternatives](https://stackoverflow.com/a/70831583).  As for `paste` specifically, it was [removed](https://developer.mozilla.org/en-US/docs/Web/API/Document/execCommand#paste) due to [security concerns](https://stackoverflow.com/a/53448633).

I [updated](https://github.com/ueberdosis/tiptap/discussions/802#discussioncomment-7300947) the clipboard extension example for Tiptap v2, though it does not support paste.
